torchaudio/models/_wavernn.py

```diff
@@ -50,56 +48,277 @@ def forward(self, x: Tensor) -> Tensor:
+            - x: :math:`(batch, freq, time)`
+            - output: :math:`(batch, n_output, time - kernel_size + 1)`
         """
 
         return self.melresnet_model(x)
+
+
+class _Stretch2d(nn.Module):
+    r"""This is a two-dimensional stretch layer. It is a block used in WaveRNN.
+
+    Args:
+        x_scale: the scale factor in x axis
+        y_scale: the scale factor in y axis
+
+    Examples::
+        >>> stretch2d = _Stretch2d(x_scale=10, y_scale=10)
+
+        >>> input = torch.rand(10, 1, 100, 512)
+        >>> output = stretch2d(input)
+    """
+
+    def __init__(self,
+                 x_scale: int,
+                 y_scale: int) -> None:
+        super().__init__()
+
+        self.x_scale = x_scale
+        self.y_scale = y_scale
+
+    def forward(self, x: Tensor) -> Tensor:
+        r"""
+        Args:
+            x: the input sequence to the _Stretch2d layer
+
+        Shape:
+            - x: :math:`(..., freq, time)`
+            - output: :math:`(..., freq * y_scale, time * x_scale)`
+        """
+
+        return x.repeat_interleave(self.y_scale, 2).repeat_interleave(self.x_scale, 3)
+
+
+class _UpsampleNetwork(nn.Module):
+    r"""This is an upsample block based on a stack of Conv2d and Strech2d layers.
+    It is a block used in WaveRNN.
+
+    Args:
+        upsample_scales: the list of upsample scales
+        n_res_block: the number of ResBlock in stack (default=10)
+        n_freq: the number of bins in a spectrogram (default=128)
+        n_hidden: the number of hidden dimensions (default=128)
+        n_output: the number of output dimensions (default=128)
+        kernel_size: the number of kernel size in the first Conv1d layer (default=5)
+
+    Examples::
+        >>> upsamplenetwork = _UpsampleNetwork(upsample_scales=[4, 4, 16],
+                                               n_res_block=10,
+                                               n_freq=128,
+                                               n_hidden=128,
+                                               n_output=128,
+                                               kernel_size=5)
+        >>> input = torch.rand(10, 128, 512)
+        >>> output = upsamplenetwork(input)
+    """
+
+    def __init__(self,
+                 upsample_scales: List[int],
+                 n_res_block: int = 10,
+                 n_freq: int = 128,
+                 n_hidden: int = 128,
+                 n_output: int = 128,
+                 kernel_size: int = 5) -> None:
+        super().__init__()
+
+        total_scale = 1
+        for upsample_scale in upsample_scales:
+            total_scale *= upsample_scale
+
+        self.indent = (kernel_size - 1) // 2 * total_scale
+        self.resnet = _MelResNet(n_res_block, n_freq, n_hidden, n_output, kernel_size)
+        self.resnet_stretch = _Stretch2d(total_scale, 1)
+
+        up_layers = []
+        for scale in upsample_scales:
+            k_size = (1, scale * 2 + 1)
+            padding = (0, scale)
```

Comment:
defining variables like this looks uncommon yet it does not improve readability.

just put variables in keyword argument and it is readable enough.

```
conv = nn.Conv2d(
    in_channels=1, out_channels=1, kernel_size= (1, scale * 2 + 1), padding=(0, scale), bias=False)
```

Reply:
Fixed, thanks.